### PR TITLE
Add upgrade to v2.3.0 notes

### PR DESCRIPTION
**Pull request by caseydavenport**
Adds upgrade instructions for KDD from < v2.3.0 to >= v2.3.0 in order to maintain NetworkPolicy behavior.

### CODE REVIEW

File: master/getting-started/kubernetes/upgrade.md [hunk 1]
```diff
@@ -167,3 +172,31 @@ To upgrade the policy controller:
...
+
+## Upgrading to v1 NetworkPolicy semantics
+
+Calico v2.3.0 when using the Kubernetes datastore driver interprets the Kubernetes `NetworkPolicy` differently
+than previous releases, as specified in [upstream Kubernetes](https://github.com/kubernetes/kubernetes/pull/39164#issue-197243974).
+
+To maintain behavior when upgrading to Calico v2.3.0+, you should follow the following steps.
+
+- In any Namespace that previously did _not_ have a "DefaultDeny" annotation:
+  - Delete any NetworkPolicy objects in that Namespace.  After upgrade, these policies will become active and my block traffic that was previously allowed.
```
Comment (robbrockbank) [inline master/getting-started/kubernetes/upgrade.md]: my -> may

File: master/getting-started/kubernetes/upgrade.md [hunk 2]
```diff
@@ -167,3 +172,31 @@ To upgrade the policy controller:
 We recommend running the policy controller as a Kubernetes Deployment with type "recreate", in which
 case upgrade can be handled entirely through the
 standard [Deployment mechanism](http://kubernetes.io/docs/user-guide/deployments/#updating-a-deployment)
+
+## Upgrading to v1 NetworkPolicy semantics
+
+Calico v2.3.0 when using the Kubernetes datastore driver interprets the Kubernetes `NetworkPolicy` differently
+than previous releases, as specified in [upstream Kubernetes](https://github.com/kubernetes/kubernetes/pull/39164#issue-197243974).
+
+To maintain behavior when upgrading to Calico v2.3.0+, you should follow the following steps.
```
Comment (robbrockbank) [inline master/getting-started/kubernetes/upgrade.md]: "follow these steps" perhaps

Comment (lwr20) [inline master/getting-started/kubernetes/upgrade.md]: "To maintain behavior when upgrading to Calico v2.3.0+, you should follow the following steps" - *when* do I do those steps?  Before upgrading?  After upgrading?  Will there be a window then default-deny isn't working?

File: master/getting-started/kubernetes/upgrade.md [hunk 3]
```diff
@@ -167,3 +172,31 @@ To upgrade the policy controller:
...
+metadata:
+  name: default-deny
+spec:
+  podSelector:
+```
+
+> **Note**:
+>
+> The above steps should be followed when upgrading to Calico v2.3.0+ using the Kubernetes
+> datastore driver, independent of the Kubernetes version being used.
```
Comment (lwr20) [inline master/getting-started/kubernetes/upgrade.md]: If you are using the Kubernetes datastore driver, you should perform the steps above when upgrading to Calico v2.3.0+, regardless of the Kubernetes version.
You do NOT need to perform the above steps if you are using the etcd datastore driver (in v2.3.x - the corresponding change is planned for the etcd datastore driver in a v2.4.x release)

File: master/getting-started/kubernetes/upgrade.md [hunk 4]
```diff
@@ -167,3 +172,31 @@ To upgrade the policy controller:
 We recommend running the policy controller as a Kubernetes Deployment with type "recreate", in which
 case upgrade can be handled entirely through the
 standard [Deployment mechanism](http://kubernetes.io/docs/user-guide/deployments/#updating-a-deployment)
+
+## Upgrading to v1 NetworkPolicy semantics
+
+Calico v2.3.0 when using the Kubernetes datastore driver interprets the Kubernetes `NetworkPolicy` differently
```
Comment (lwr20) [inline master/getting-started/kubernetes/upgrade.md]: Calico v2.3.0 (when using the Kubernetes datastore driver) interprets the Kubernetes `NetworkPolicy` differently

File: master/getting-started/kubernetes/upgrade.md [hunk 5]
```diff
@@ -167,3 +172,31 @@ To upgrade the policy controller:
...
+than previous releases, as specified in [upstream Kubernetes](https://github.com/kubernetes/kubernetes/pull/39164#issue-197243974).
```
Comment (lwr20) [inline master/getting-started/kubernetes/upgrade.md]: If your Calico deployment uses the *etcd datastore driver*, there is no change in behaviour in v2.3.0+ (this change is planned for a 2.4.0 release).

File: v2.3/getting-started/kubernetes/upgrade.md [hunk 6]
```diff
@@ -168,3 +173,31 @@ To upgrade the policy controller:
...
+
+## Upgrading to v1 NetworkPolicy semantics
+
+Calico v2.3.0 when using the Kubernetes datastore driver interprets the Kubernetes `NetworkPolicy` differently
+than previous releases, as specified in [upstream Kubernetes](https://github.com/kubernetes/kubernetes/pull/39164#issue-197243974).
+
+To maintain behavior when upgrading to Calico v2.3.0+, you should follow the following steps.
+
+- In any Namespace that previously did _not_ have a "DefaultDeny" annotation:
+  - Delete any NetworkPolicy objects in that Namespace.  After upgrade, these policies will become active and my block traffic that was previously allowed.
```
Comment (lwr20) [inline v2.3/getting-started/kubernetes/upgrade.md]: I assume you're going to port all the changes from master into here, right?

Comment (caseydavenport) [inline v2.3/getting-started/kubernetes/upgrade.md]: Which changes from master? This should be identical to what is in master.

File: master/getting-started/kubernetes/upgrade.md (hunk 3 above)
Comment (caseydavenport) [inline master/getting-started/kubernetes/upgrade.md]: We already say that in a bunch of other places in this doc.

**Comment by caseydavenport** [comment]
@lwr20 @robbrockbank Ok, I think I've addressed your comments. Ready for another look.

File: master/getting-started/kubernetes/upgrade.md [hunk 7]
```diff
@@ -167,3 +172,33 @@ To upgrade the policy controller:
...
 standard [Deployment mechanism](http://kubernetes.io/docs/user-guide/deployments/#updating-a-deployment)
+
+## Upgrading to v1 NetworkPolicy semantics
+
+Calico v2.3.0 (when using the Kubernetes datastore driver) interprets the Kubernetes `NetworkPolicy` differently
+than previous releases, as specified in [upstream Kubernetes](https://github.com/kubernetes/kubernetes/pull/39164#issue-197243974).  If your
+Calico deployment uses the etcd datastore driver, there is no change in behavior in v2.3.0+ (this change is planned for the v2.4.0 release).
+
+To maintain behavior when upgrading to Calico v2.3.0+, you should follow these steps prior to upgrading Calico to ensure policy is
+enforced the same throughout the upgrade process.
```
Comment (lwr20) [inline master/getting-started/kubernetes/upgrade.md]: "ensure policy is enforced the same" feels grammatically weird to me.  Perhaps "ensure policy enforcement does not change"?

File: master/getting-started/kubernetes/upgrade.md [hunk 8]
```diff
@@ -167,3 +172,33 @@ To upgrade the policy controller:
 We recommend running the policy controller as a Kubernetes Deployment with type "recreate", in which
 case upgrade can be handled entirely through the
 standard [Deployment mechanism](http://kubernetes.io/docs/user-guide/deployments/#updating-a-deployment)
+
+## Upgrading to v1 NetworkPolicy semantics
+
+Calico v2.3.0 (when using the Kubernetes datastore driver) interprets the Kubernetes `NetworkPolicy` differently
+than previous releases, as specified in [upstream Kubernetes](https://github.com/kubernetes/kubernetes/pull/39164#issue-197243974).  If your
+Calico deployment uses the etcd datastore driver, there is no change in behavior in v2.3.0+ (this change is planned for the v2.4.0 release).
```
Comment (lwr20) [inline master/getting-started/kubernetes/upgrade.md]: This feels like the most important couple of statements in the entire doc.  Is it possible to make this huge and bold and in a big red box or something to draw people's attention to it?

File: v2.3/getting-started/kubernetes/upgrade.md [hunk 9]
```diff
@@ -168,3 +173,33 @@ To upgrade the policy controller:
...
 standard [Deployment mechanism](http://kubernetes.io/docs/user-guide/deployments/#updating-a-deployment)
+
+## Upgrading to v1 NetworkPolicy semantics
+
+Calico v2.3.0 (when using the Kubernetes datastore driver) interprets the Kubernetes `NetworkPolicy` differently
+than previous releases, as specified in [upstream Kubernetes](https://github.com/kubernetes/kubernetes/pull/39164#issue-197243974).  If your
+Calico deployment uses the etcd datastore driver, there is no change in behavior in v2.3.0+ (this change is planned for the v2.4.0 release).
+
+To maintain behavior when upgrading to Calico v2.3.0+, you should follow these steps prior to upgrading Calico to ensure policy is
+enforced the same throughout the upgrade process.
```
Comment (lwr20) [inline v2.3/getting-started/kubernetes/upgrade.md]: Same clunky wording as above.

**Pull request merged**